book/src/concepts/types.md

```diff
@@ -7,73 +7,68 @@ SPDX-License-Identifier: MIT OR Apache-2.0
 
 # Types
 
-## Primitive Trivial Types
+CXX-Qt any types supported by CXX, these can be used in properties, invokables, and signals.
 
-These types can be used for properties, parameters or return types in invokables, and parameters in signals without any conversion.
+Note that there is missing support for a few CXX types https://github.com/KDAB/cxx-qt/issues/328
 
-They appear as their normal types on both the C++ and Rust sides of the bridge.
+## `cxx-qt-lib` Types
 
-| Rust Type | C++ Type |
-|-----------|----------|
-| bool      | bool     |
-| f32       | float    |
-| f64       | double   |
-| i8        | qint8    |
-| i16       | qint16   |
-| i32       | qint32   |
-| u8        | quint8   |
-| u16       | quint16  |
-| u32       | quint32  |
+The `cxx-qt-lib` crate provides common types from Qt as CXX supported types.
 
-TODO: Note that u64 / quint64 is not supported currently ( [https://github.com/KDAB/cxx-qt/issues/36](https://github.com/KDAB/cxx-qt/issues/36) ).
+Use the [`cxx-qt-lib` Docs](https://docs.rs/cxx-qt-lib/latest/cxx_qt_lib/) to explore the available types.
 
-## Custom Types
+## Defining a Custom Type
 
-These types are custom and require special treatment when traversing the bridge, to assist with traversing the bridge we have provided helper types in the cxx_qt_lib crate.
+Any types that are valid CXX types should be usable with CXX-Qt as well.
 
-Within these custom types there are two kinds to consider
+Note that the same rules apply as CXX, so a type must be trivial to pass around as value.
+If they are opaque then references or pointers must be used.
 
-  * Trivial
-  * Opaque
+For examples of how to wrap Qt objects explore the `cxx-qt-lib` source code.
 
-### Custom Trivial Types
+## Opaque Type Conversions
 
-Custom trivial types, like primitive trival types, can be used for properties, parameters or return types in invokables, and parameters in signals without any conversion.
+If your type can't be marked as trivial for CXX, but the Qt API needs to take the type by value (eg `T` rather than `const T&` or `::std::unique_ptr<T>`).
 
-On the rust side they appear as the cxx_qt_lib helper type.
+Then you can define [type conversions](type-conversions.md) in C++ that convert between pointers / references and values.
+Then when using properties, invokables, or signals an attribute in the macro can be used.
 
-Note that when they are used as a parameter type in invokables they should be passed as a reference, eg `pointf: &QPointF`, and when they are a property or return type they should be a value, eg `QPointF`.
+### Properties
 
-| Rust Type | C++ Type |
-|-----------|----------|
-| cxx_qt_lib::QDate | QDate |
-| cxx_qt_lib::QPoint | QPoint |
-| cxx_qt_lib::QPointF | QPointF |
-| cxx_qt_lib::QRect | QRect |
-| cxx_qt_lib::QRectF | QRectF |
-| cxx_qt_lib::QTime | QTime |
+In the following example using `cxx_type` means that `OpaqueExampleType` will be the type of the `Q_PROPERTY`, getter, and setter in C++.
```

Comment:
I'm confused. There's no `cxx_type` in the example code.

Reply:
In the `cxx_type = "OpaqueExampleType" ` in the `#[qproperty(cxx_type = "OpaqueExampleType")]` line